src/js/node/http2.ts

```diff
@@ -7,27 +7,963 @@ const { hideFromStack, throwNotImplemented } = require("internal/shared");
+    throw new ERR_HTTP2_INVALID_HEADER_VALUE(value, name);
+  }
+  if (!isConnectionHeaderAllowed(name, value)) {
+    connectionHeaderMessageWarn();
+  }
+}
+
+hideFromStack(assertValidHeader);
+
+class Http2ServerRequest extends Readable {
+  constructor(stream, headers, options, rawHeaders) {
+    super({ autoDestroy: false, ...options });
+    this[kState] = {
+      closed: false,
+      didRead: false,
+    };
+    // Headers in HTTP/1 are not initialized using Object.create(null) which,
+    // although preferable, would simply break too much code. Ergo header
+    // initialization using Object.create(null) in HTTP/2 is intentional.
+    this[kHeaders] = headers;
+    this[kRawHeaders] = rawHeaders;
+    this[kTrailers] = {};
+    this[kRawTrailers] = [];
+    this[kStream] = stream;
+    this[kAborted] = false;
+    stream[kRequest] = this;
+
+    // Pause the stream..
+    stream.on("trailers", onStreamTrailers);
+    stream.on("end", onStreamEnd);
+    stream.on("error", onStreamError);
+    stream.on("aborted", onStreamAbortedRequest);
+    stream.on("close", onStreamCloseRequest);
+    stream.on("timeout", onStreamTimeout);
+    this.on("pause", onRequestPause);
+    this.on("resume", onRequestResume);
+  }
+
+  get aborted() {
+    return this[kAborted];
+  }
+
+  get complete() {
+    return this[kAborted] || this.readableEnded || this[kState].closed || this[kStream].destroyed;
+  }
+
+  get stream() {
+    return this[kStream];
+  }
+
+  get headers() {
+    return this[kHeaders];
+  }
+
+  get rawHeaders() {
+    return this[kRawHeaders];
+  }
+
+  get trailers() {
+    return this[kTrailers];
+  }
+
+  get rawTrailers() {
+    return this[kRawTrailers];
+  }
+
+  get httpVersionMajor() {
+    return 2;
+  }
+
+  get httpVersionMinor() {
+    return 0;
+  }
+
+  get httpVersion() {
+    return "2.0";
+  }
+
+  get socket() {
+    return this[kStream]?.[bunHTTP2Session]?.socket;
+  }
+
+  get connection() {
+    return this.socket;
+  }
+
+  _read(nread) {
+    const state = this[kState];
+    if (!state.didRead) {
+      state.didRead = true;
+      this[kStream].on("data", onStreamData);
+    } else {
+      process.nextTick(resumeStream, this[kStream]);
+    }
+  }
+
+  get method() {
+    return this[kHeaders][HTTP2_HEADER_METHOD];
+  }
+
+  set method(method) {
+    validateString(method, "method");
+    if (StringPrototypeTrim(method) === "") throw new ERR_INVALID_ARG_VALUE("method", method);
+
+    this[kHeaders][HTTP2_HEADER_METHOD] = method;
+  }
+
+  get authority() {
+    return getAuthority(this[kHeaders]);
+  }
+
+  get scheme() {
+    return this[kHeaders][HTTP2_HEADER_SCHEME];
+  }
+
+  get url() {
+    return this[kHeaders][HTTP2_HEADER_PATH];
+  }
+
+  set url(url) {
+    this[kHeaders][HTTP2_HEADER_PATH] = url;
+  }
+
+  setTimeout(msecs, callback) {
+    if (!this[kState].closed) this[kStream].setTimeout(msecs, callback);
+    return this;
+  }
+}
+class Http2ServerResponse extends Stream {
```

Comment:
Is this a class in Node?

Reply:
It is https://github.com/nodejs/node/blob/4dc76a990f0037401b6bf8c982d7401c2533f23b/lib/internal/http2/compat.js#L468-L469